Sources/Sentry/SentryUIEventTracker.m

```diff
@@ -0,0 +1,159 @@
+                if (target == nil || sender == nil) {
+                    return;
+                }
+
+                if (![sender isKindOfClass:[UIButton class]]
+                    && ![sender isKindOfClass:[UISegmentedControl class]]
+                    && ![sender isKindOfClass:[UIPageControl class]]) {
+                    return;
+                }
+
+                UIView *view = sender;
+                BOOL sameView = self.activeView != nil && view == self.activeView;
+                if (sameView) {
+                    [self.activeTransaction dispatchIdleTimeout];
+                    return;
+                }
+
+                [self.activeTransaction finish];
+
+                NSString *transactionName = [self getTransactionName:action
+                                                              target:target
+                                                                view:view];
+
+                SentryTransactionContext *context = [[SentryTransactionContext alloc]
+                    initWithName:transactionName
+                       operation:SentrySpanOperationUIActionClick];
+
+                BOOL ongoingScreenLoadTransaction = span != nil &&
+                    [span.context.operation isEqualToString:SentrySpanOperationUILoad];
+                BOOL ongoingManualTransaction = span != nil
+                    && ![span.context.operation isEqualToString:SentrySpanOperationUILoad]
+                    && ![span.context.operation containsString:SentrySpanOperationUIAction];
+
+                BOOL bindToScope = !ongoingScreenLoadTransaction && !ongoingManualTransaction;
+                SentryTracer *transaction =
+                    [SentrySDK.currentHub startTransactionWithContext:context
+                                                          bindToScope:bindToScope
+                                                customSamplingContext:@{}
+                                                          idleTimeout:defaultIdleTransactionTimeout
+                                                 dispatchQueueWrapper:self.dispatchQueueWrapper];
+
+                transaction.finishCallback = ^(void) {
+                    self.activeTransaction = nil;
+                    self.activeView = nil;
+                };
+
+                self.activeTransaction = transaction;
+                self.activeView = view;
+                self.activeEventType = event.type;
+            }];
+        }
+                   forKey:SentryUIEventTrackerSwizzleSendAction];
+}
+
+- (void)stop
+{
+    [self.swizzleWrapper removeSwizzleSendActionForKey:SentryUIEventTrackerSwizzleSendAction];
+}
+
+- (NSString *)getTransactionName:(NSString *)action target:(id)target view:(UIView *)element
+{
+    NSString *targetClass = NSStringFromClass([target class]);
+
+    if (element.accessibilityIdentifier) {
+        return [NSString stringWithFormat:@"%@.%@", targetClass, element.accessibilityIdentifier];
+    }
```

Comment:
m: I believe the action name is way more important than the control accessibilitiyIdentirifer. 
But we could use both, something like `class.action()  {caller: accessibilityIdentifier}`

Because the control's action can be changed programmatically. IMO it is more relevant to know what code was executed than what control called the execution.

We also include the control Identifier because more than one control can target the same function.

Reply:
That's an excellent point. If we add the `{caller: accessibilityIdentifier}` to the transaction name, we would get different transactions depending on the caller. I think the transaction should be the same no matter who is calling it. We can attach the caller somewhere else as for example additional data.